lib/galaxy/model/store/__init__.py

```diff
@@ -37,14 +38,32 @@
 GALAXY_EXPORT_VERSION = "2"
 
 
+class ImportDiscardedDataType(Enum):
+    # Don't allow discarded 'okay' datasets on import, datasets will be marked deleted.
+    FORBID = 'forbid'
+    # Allow datasets to be imported as experimental DISCARDED datasets that are not deleted if file data unavailable.
+    ALLOW = 'allow'
```

Comment:
I am very confused by this option. This does need to be set to deleted & purged on import, right ?

Reply:
The whole point here is invest in this DISCARDED type that is for non-deleted datasets that don't have a physical file. So you can browse the history in the UI and such and see the job properties, metadata, etc.. but not have to have the file - using Galaxy as a provenance browser for the imported histories for instance.

Since this isn't obvious though, I'll close this PR and see if I can handle DISCARDED and differed datasets at the same time. There are similarities and differences there and I guess we should explore them before investing in these concepts.